kim/src/BJ2812.java

```diff
@@ -0,0 +1,45 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Stack;
+import java.util.StringTokenizer;
+
+public class BJ2812 {
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        int n = Integer.parseInt(st.nextToken());
+        int k = Integer.parseInt(st.nextToken());
+        String s = br.readLine();
+
+        Stack<Character> stack = new Stack<>();
+        stack.push(s.charAt(0));
+        int cnt = 0;
+
+        for (int i = 1; i < n; i++) {
+            char c = s.charAt(i);
+
+            while (!stack.isEmpty() && cnt < k) {
+                if (stack.peek() >= c)
+                    break;
+                // 전이 더 작으면 전에거를 제거
+                stack.pop();
+                cnt++;
+            }
+            // 아니면 그냥 이어붙여줌
+            stack.push(c);
```

Comment:
이 조건식은 이런식으로 짜면 좀더 깔끔해집니다.
```suggestion
            while (!stack.isEmpty() && cnt < k && stack.peek() < c) {
                stack.pop();
                cnt++;
            }
            stack.push(c);
```

Reply:
감사합니다! 수정 완료했습니다
제 코드 짤 때는 이런게 잘 안보이드라구여ㅠ